nav2_mppi_controller/src/optimizer.cpp

```diff
@@ -306,78 +292,107 @@ void Optimizer::propagateStateVelocitiesFromInitials(
 }
 
 void Optimizer::integrateStateVelocities(
-  xt::xtensor<float, 2> & trajectory,
-  const xt::xtensor<float, 2> & sequence) const
+  Eigen::ArrayXXf & trajectory,
+  const Eigen::ArrayXXf & sequence) const
 {
   float initial_yaw = static_cast<float>(tf2::getYaw(state_.pose.pose.orientation));
+  const float & dt_time = settings_.model_dt;
 
-  const auto vx = xt::view(sequence, xt::all(), 0);
-  const auto wz = xt::view(sequence, xt::all(), 1);
+  const auto vx = sequence.col(0);
+  const auto wz = sequence.col(1);
 
-  auto traj_x = xt::view(trajectory, xt::all(), 0);
-  auto traj_y = xt::view(trajectory, xt::all(), 1);
-  auto traj_yaws = xt::view(trajectory, xt::all(), 2);
+  auto traj_x = trajectory.col(0);
```

Comment:
Are these copies or references?

Reply:
As I understand, these variables are abstract expressions representing the block view of the Eigen Array and storing references to the original variable. If we do eval(), it will copy the column data to the new variable.
[reference](https://eigen.tuxfamily.org/dox/TopicPitfalls.html#:~:text=In%20this%20example,.)